lib/ignore/ignore.go

```diff
@@ -99,16 +116,16 @@ func (m *Matcher) Parse(r io.Reader, file string) error {
 	return err
 }
 
-func (m *Matcher) Match(file string) (result bool) {
+func (m *Matcher) Match(file string) (result Result) {
 	if m == nil {
-		return false
+		return Result{false, false}
```

Comment:
I'd generally prefer to see these as `Result{include: false, deletable: false}` to be explicit about what the various booleans are, or maybe just a `Result{}` as the zero value is a non-deletable non-included non-match, or as a named constant as we return it in a couple of places. But this is just a style opinion.